configure.ac

```diff
@@ -251,29 +251,38 @@ AC_TYPE_INT16_T
 AC_TYPE_INT32_T
 AC_TYPE_INT64_T
 
+PKG_CHECK_MODULES([libqb], [libqb],
+		  [if test "x$libqb_PREFIX" = "x"; then libqb_PREFIX=`pkg-config --variable=prefix libqb`; fi])
+
+if test "x$libqb_PREFIX" = "x"; then
+	AC_MSG_ERROR([m4_text_wrap([
+  Package requirements (libqb) were not met.
+  This generally happens when configure is unable to determine libqb install
+  prefix.
+
+  Consider adjusting the PKG_CONFIG_PATH environment variable if you installed
+  software in a non-standard prefix.
+
+  Alternatively, you may set the environment variable libqb_PREFIX
+  to avoid the need to call pkg-config.
+  ], [  ], [   ], [76])])
+fi
+
 if test "x$enable_man" = "xyes"; then
 	AC_ARG_VAR([DOXYGEN], [override doxygen executable])
 	AC_CHECK_PROGS([DOXYGEN], [doxygen], [no])
 	if test "x$DOXYGEN" = xno; then
 		AC_MSG_ERROR(["Doxygen command not found"])
 	fi
-	# required by doxyxml to build man pages dynamically
-	# Don't let AC_PROC_CC (invoked by AX_PROG_CC_FOR_BUILD) replace
-	# undefined CFLAGS_FOR_BUILD with -g -O2, overriding our special OPT_CFLAGS.
-	: ${CFLAGS_FOR_BUILD=""}
-	AX_PROG_CC_FOR_BUILD
-	saved_PKG_CONFIG="$PKG_CONFIG"
-	saved_ac_cv_path_PKG_CONFIG="$ac_cv_path_PKG_CONFIG"
-	unset PKG_CONFIG ac_cv_path_PKG_CONFIG
-	AC_PATH_PROG([PKG_CONFIG], [pkg-config])
-	PKG_CHECK_MODULES([libqb_BUILD], [libqb])
-	PKG_CHECK_MODULES([libxml_BUILD], [libxml-2.0])
-	PKG_CONFIG="$saved_PKG_CONFIG"
-	ac_cv_path_PKG_CONFIG="$saved_ac_cv_path_PKG_CONFIG"
+	AC_CHECK_PROGS([DOXYGEN2MAN], [doxygen2man], [no], [$libqb_PREFIX/bin])
```

Comment:
Here you conflate the build-system binary (found by `AC_CHECK_PROGS`) with a host-system variable (`libqb_PREFIX`). Actually, using the plain `pkg-config` invocation above makes this work out, because `PKG_CHECK_MODULES` uses the host-system `pkg-config`, while the direct invocation calls the build-system `pkg-config`, but this seems more a happenstance than calculated design due to the lack of comments. :) And also makes the usual variable overrides ineffective.

Reply:
Yeah I was not convinced about this either.. and still exploring the limitations of what I have created with too little coffee in my veins :P